react/features/filmstrip/functions.web.js

```diff
@@ -697,11 +696,10 @@ export function getPinnedActiveParticipants(state) {
 export function shouldDisplayStageFilmstrip(state, minParticipantCount = 2) {
     const { activeParticipants } = state['features/filmstrip'];
     const { remoteScreenShares } = state['features/video-layout'];
-    const currentLayout = getCurrentLayout(state);
```

Comment:
I'm wondering is it possible to be in tile view and this function to return true? Shouldn't we check for this?

Reply:
This function is only called when we're not in TILE_VIEW. The layout check was removed because we now use this function in order to determine layout

Reply:
IMHO checking if we are not in tile view won't hurt and I suppose that it can be used even when we are tile view in future.

Reply:
Yes, but to check if we're tile view we need to call getCurrentLayout which calls this function, so it would cause an infinite loop

Reply:
Not really. You can call `shouldDisplayTileView`.

Reply:
I checked again and got a little bit confused about the `shouldDisplayStageFilmstrip` function.

Let me verify something:
 - It looks to me that `shouldDisplayStageFilmstrip` in the `onClick` method of `Thumbnail` and also in the middleware which handles the dominant speaker changes needs return true even in tile view so that we can keep the`activeParticipants` in redux up to date. Is this true?
 - I'm not sure what is the situation with the one in `_electParticipantInLargeVideo`. Correct me if I'm wrong but IMHO this one is definitely called in tile view. Should we check there if we are not in tile view?
 - It seems all the other usages of `shouldDisplayStageFilmstrip` are not necessary. I've added additional comments for those.

I think if it turns out that we need `shouldDisplayStageFilmstrip` to return true even when we are in tile view then we need to change its name to something that indicates the stage filmstrip may not be displayed at the moment.

One more question on the matter:
 - I see that `getPinnedParticipant` is used when `shouldDisplayStageFilmstrip` is true in `_electParticipantInLargeVideo` which can be called even in tile view and also when there are multiple participants on stage. I'm wondering what would this lead to? Do you think there will be some issues? Do we need to elect something for large video in first place?

Reply:
- Yes. Although I need to add a check on dominant speaker change that we're not in tile view

- It is called in tile view, but for no reason. That function is used to determine which participant should be displayed in VERTICAL_FILMSTRIP view. I will try to change it so that it only works in vertical_filmstrip view.

- They're necessary. Responded to the comments.

- No, we don't need to elect something for large video, like I said above I'll update that

Reply:
I do understand the confusion and I think that `shouldDisplayStageFilmstrip` is not the best name. I'll try to refactor it to make it more clear

Reply:
I think I found one more place where we require `shouldDisplayStageFilmstrip` to return `true` even in tile view. Left you a comment there.

Also again I think if we won't check for `shouldDisplayTileView` in `shouldDisplayStageFilmstrip` we should definitely change the name because I see myself using it to check if we display the stage filmstrip because of the name.

Reply:
I agree it's not the best name. I'll change it